### PR TITLE
feat: add segment `IN` operator

### PR DESCRIPTION
**Pull request by khvn26**

### CODE REVIEW

File: tests/Engine/Unit/Segments/SegmentModelsTest.php [hunk 1]
```diff
@@ -59,6 +59,16 @@ public function conditionParametersTraitValues()
...
+            [SegmentConditions::IN, 'foo', '', false],
+            [SegmentConditions::IN, 'ba', 'foo,bar', false],
+            [SegmentConditions::IN, 'foo', 'foo,bar', true],
+            [SegmentConditions::IN, 'bar', 'foo,bar', true],
+            [SegmentConditions::IN, 'foo', 'foo', true],
+            [SegmentConditions::IN, 1, '1,2,3,4', true],
+            [SegmentConditions::IN, 1, '', false],
+            [SegmentConditions::IN, 1, '1', true],
+            [SegmentConditions::IN, 1.5, '1.5', false],
+            [SegmentConditions::IN, false, 'false', false],
```
Comment (matthewelwell) [inline tests/Engine/Unit/Segments/SegmentModelsTest.php:71]: We should perhaps label these 2 with a comment? Or the logic itself. It's not immediately obvious why we're not handling floats / booleans (although I fully agree with the reasoning).

Comment (khvn26) [reply tests/Engine/Unit/Segments/SegmentModelsTest.php:71]: Do you think it's a good place to document this? My plan was to update the documentation with the proper explanation.

Comment (matthewelwell) [reply tests/Engine/Unit/Segments/SegmentModelsTest.php:71]: I think both?

Comment (khvn26) [reply tests/Engine/Unit/Segments/SegmentModelsTest.php:73]: Done.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**